### DISCUSSION TITLE
Blender import addon not working

### DISCUSSION THREAD

**Discussion by 311-code** [thygate/stable-diffusion-webui-depthmap-script#116] [General]
I just installed newest blender and tried @LadyPoly's depthmap import addon. After install it does not show "import depth map" in the import options. I have installed many addons before so not sure if it just broke with newest version of blender? It doesn't even show up in the list of addons after install.

My goal here was to import the ply files and textures to blender then export to FBX, then import it all to Unity for seeing stable diffusion 3D environment in VR with a Quest 2.

I did try to just export the ply as an fbx to blender then apply the unlit texture but it doesn't work. It's like it has no uv maps.

**Comment by thygate** [comment]
The ladypoly blender addon imports a texture and depthmap and uses it to displace a plane, you could still do this manually ..
You don't need it to import the PLY files either (see bottom of readme).

I can't help you with the blender plugin. When you say latest version, it would be helpful to include the actual version for future references.

**Comment by 311-code** [comment]
Thanks, I just did blender 3.4.50, I went all the way back to 3.3.0 and addon still does not show up on that older one either sadly.

Oh you don't need to import the PLY? I am looking through the main page here and don't see the info for this for displacing on a plane. I just realized blender has a [VR addon](https://www.youtube.com/watch?v=ge7Dl6L8lUE) built you can turn on, in so I can avoid Unity altogether!

**Comment by thygate** [comment]
The blender VR addon has a lot of lag and needs to do a lot of redrawing when you move your head. Not the best experience imo.

I was referring to the instructions on how to import the ply in blender :

To view the 3D-inpainted mesh in blender:

    import the PLY file
    Set camera to origin (0, 0, 0) and pointing up to align it with the mesh
    adjust camera FOV to match the mesh
    Add a 'Color Attribute' Node and connect it to the shader color input

I just tried to install the ladypoly blender addon in blender 3.4.1 and it works fine, just as it did on 3.3

Edit/Preferences
Addons
Install
Browse zip file
done

**Comment by 311-code** [comment]
I was doing the wrong .zip doh! Thanks it's working. Okay so when I import the combined depthmap and use that view in VR it was very choppy. Then I closed the automatic1111 cmd window and now it's buttery smooth. This is amazing right now!

Thanks, for your help.

**Comment by Ladypoly** [comment]
Unfortunately the ply for the inpainted mesh is very high poly and due to its heavy perspective mesh displaced nature it is not easy to unwrap and bake the vertex colours to a texture. 
My plan was to load it into the quest too but after spending a whole weekend with many crashes I gave up

**Comment by AugmentedRealityCat** [comment]
> I did try to just export the ply as an fbx to blender then apply the unlit texture but it doesn't work. It's like it has no uv maps.

- You can **import the PLY in blender** - you don't need it to be an FBX.
- You don't need to apply the image you extracted the depth map from as a texture.
- You cannot use UV coordinates anyways because **the PLY model doesn't have any UV coordinates.**
- All the color information, including the inpainted areas, are **stored as Vertex Colors** in a Vertex Map. 
- Since there are parts of the 3d scene that have been extended at the back and inpainted by the process, it would not be possible to use your source picture as a texture map because  the inpainted parts are not visible in your source image anyways - this is because those parts are hidden behind the objects we see on your image.
- To see the vertex colors you have to change the viewer mode in Blender and to apply a material that will show them. You can also export the PLY model in another format, like FBX, and if you set the paramters correctly the Vertex Color map will get exported as well. That's how I bring those models in Cinema4d, which is the 3d app I use.
